src/DateTimeParser.jl

```diff
@@ -0,0 +1,512 @@
+            end
+        else
+            if ymd[1] > 31 || (yearfirst && ymd[2] <= 12 && ymd[3] <= 31)
+                # 99-01-01
+                res["year"], res["month"], res["day"] = ymd
+            elseif ymd[1] > 12 || (dayfirst && ymd[2] <= 12)
+                # 13-01-01
+                res["day"], res["month"], res["year"] = ymd
+            else
+                # 01-13-01
+                res["month"], res["day"], res["year"] = ymd
+            end
+        end
+    end
+
+    # Fill in default values if none exits
+    if !haskey(res, "year") && !haskey(res, "month") && !haskey(res, "day")
+        res["year"], res["month"], res["day"] = Dates.yearmonthday(default)
+    else
+        if haskey(res, "year")
+            res["year"] = convertyear(res["year"])
+        else
+            res["year"] = year(default)
+        end
+        if !haskey(res, "month")
+            res["month"] = 1
+        end
+        if !haskey(res, "day")
+            res["day"] = 1
+        end
+    end
+    if !haskey(res, "hour") && !haskey(res, "minute") && !haskey(res, "second") && !haskey(res, "millisecond")
+        res["hour"] = hour(default)
+        res["minute"] = minute(default)
+        res["second"] = second(default)
+        res["millisecond"] = millisecond(default)
+    else
+        if !haskey(res, "hour")
+            res["hour"] = 0
+        end
+        if !haskey(res, "minute")
+            res["minute"] = 0
+        end
+        if !haskey(res, "second")
+            res["second"] = 0
+        end
+        if !haskey(res, "millisecond")
+            res["millisecond"] = 0
+        end
+    end
+    # determine timezone
+    if !haskey(res, "tzname") && !haskey(res, "tzoffset")
+        res["timezone"] = default.timezone
+    elseif !haskey(res, "tzoffset")
+        if haskey(timezone_infos, res["tzname"])
+            res["timezone"] = timezone_infos[res["tzname"]]
+        elseif res["tzname"] in TimeZones.timezone_names()
+            res["timezone"] = TimeZone(res["tzname"])
+        elseif lowercase(res["tzname"]) in UTCZONE
+            res["timezone"] = TimeZone("utc")
+        else
+            error("Failed to parse date")
+        end
+    elseif !haskey(res, "tzname")
+        res["timezone"] = FixedTimeZone("local",res["tzoffset"])
+    else
+        res["timezone"] = FixedTimeZone(res["tzname"], res["tzoffset"])
+    end
+
+    return ZonedDateTime(DateTime(res["year"], res["month"], res["day"], res["hour"],
+            res["minute"], res["second"], res["millisecond"]), res["timezone"])
+end
+
+# converts a 2 digit year to a 4 didgit one near year 2000 (e.g. 95 becomes 1995)
+function convertyear(year::Int)
+    if year > 99
+        return year
+    elseif year < 50
+        return year + 2000
+    else
+        return year + 1900
+    end
+end
+
+function _parsedatetokens(input::AbstractString)
+    tokens = []
+    regex = r"^(?P<token>(\d+\.\d+(?=[^\.\d]|$))|(\d+)|(((?=[^\d])\w)+))(?P<extra>.*)$"
```

Comment:
Interesting. The julia docs does it with a P: https://julia.readthedocs.org/en/latest/manual/strings/#regular-expressions